### PR TITLE
Support ES 7.7.0

### PR DESCRIPTION
**Pull request by sujithvm**
*Description of changes:*
Support for ES 7.7

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by vrozov** [comment]
Please see https://github.com/vrozov/security/tree/opendistro-7.7

File: src/main/java/com/amazon/opendistroforelasticsearch/security/dlic/rest/api/ActionGroupsApiAction.java [hunk 1]
```diff
@@ -16,6 +16,8 @@
 package com.amazon.opendistroforelasticsearch.security.dlic.rest.api;
 
 import java.nio.file.Path;
+import java.util.Arrays;
```
Comment (vrozov) [inline src/main/java/com/amazon/opendistroforelasticsearch/security/dlic/rest/api/ActionGroupsApiAction.java:19]: nit: unused import.

File: src/main/java/com/amazon/opendistroforelasticsearch/security/dlic/rest/api/ActionGroupsApiAction.java [hunk 2]
```diff
@@ -36,8 +38,28 @@
 import com.amazon.opendistroforelasticsearch.security.securityconf.impl.CType;
 import com.amazon.opendistroforelasticsearch.security.ssl.transport.PrincipalExtractor;
 
+import com.google.common.collect.ImmutableList;
+
+
```
Comment (vrozov) [inline src/main/java/com/amazon/opendistroforelasticsearch/security/dlic/rest/api/ActionGroupsApiAction.java:43]: nit: extra empty line.

File: src/main/java/com/amazon/opendistroforelasticsearch/security/dlic/rest/api/FlushCacheApiAction.java [hunk 3]
```diff
@@ -44,7 +44,16 @@
 import com.amazon.opendistroforelasticsearch.security.securityconf.impl.CType;
 import com.amazon.opendistroforelasticsearch.security.ssl.transport.PrincipalExtractor;
 
+import com.google.common.collect.ImmutableList;
+
+
```
Comment (vrozov) [inline src/main/java/com/amazon/opendistroforelasticsearch/security/dlic/rest/api/FlushCacheApiAction.java:49]: nit: extra empty line.

File: src/main/java/com/amazon/opendistroforelasticsearch/security/dlic/rest/api/FlushCacheApiAction.java [hunk 4]
```diff
@@ -134,5 +140,4 @@ protected CType getConfigName() {
 	protected void consumeParameters(final RestRequest request) {
 		// not needed
 	}
-
 }
```
Comment (vrozov) [inline src/main/java/com/amazon/opendistroforelasticsearch/security/dlic/rest/api/FlushCacheApiAction.java:143]: nit: add <LF>.

File: src/main/java/com/amazon/opendistroforelasticsearch/security/dlic/rest/api/InternalUsersApiAction.java [hunk 5]
```diff
@@ -45,10 +45,28 @@
 
 import java.io.IOException;
 import java.nio.file.Path;
+import java.util.List;
+
+import com.google.common.collect.ImmutableList;
+
```
Comment (vrozov) [inline src/main/java/com/amazon/opendistroforelasticsearch/security/dlic/rest/api/InternalUsersApiAction.java:51]: nit: delete extra empty line.

File: src/main/java/com/amazon/opendistroforelasticsearch/security/dlic/rest/api/OpenDistroSecurityConfigAction.java [hunk 6]
```diff
@@ -58,21 +63,18 @@ public OpenDistroSecurityConfigAction(final Settings settings, final Path config
     }
 
     @Override
-    protected void registerHandlers(RestController controller, Settings settings) {
-        controller.registerHandler(Method.GET, "/_opendistro/_security/api/securityconfig/", this);
-
-        //controller.registerHandler(Method.GET, "/_opendistro/_security/api/config/", this);
+    public List<Route> routes() {
+        List<Route> routes = new ArrayList<>();
```
Comment (vrozov) [inline src/main/java/com/amazon/opendistroforelasticsearch/security/dlic/rest/api/OpenDistroSecurityConfigAction.java:67]: Can static `routes` be used as there is already `allowPutOrPatch` flag that filters not implemented requests.

File: src/main/java/com/amazon/opendistroforelasticsearch/security/dlic/rest/api/PermissionsInfoAction.java [hunk 7]
```diff
@@ -45,10 +46,14 @@
 import com.amazon.opendistroforelasticsearch.security.support.ConfigConstants;
 import com.amazon.opendistroforelasticsearch.security.user.User;
 
+
```
Comment (vrozov) [inline src/main/java/com/amazon/opendistroforelasticsearch/security/dlic/rest/api/PermissionsInfoAction.java:49]: nit: delete extra empty line

File: src/main/java/com/amazon/opendistroforelasticsearch/security/dlic/rest/api/RolesApiAction.java [hunk 8]
```diff
@@ -16,6 +16,8 @@
 package com.amazon.opendistroforelasticsearch.security.dlic.rest.api;
 
 import java.nio.file.Path;
+import java.util.Arrays;
```
Comment (vrozov) [inline src/main/java/com/amazon/opendistroforelasticsearch/security/dlic/rest/api/RolesApiAction.java:19]: nit: unused import.

File: src/main/java/com/amazon/opendistroforelasticsearch/security/dlic/rest/api/RolesMappingApiAction.java [hunk 9]
```diff
@@ -15,8 +15,14 @@
 
 package com.amazon.opendistroforelasticsearch.security.dlic.rest.api;
 
-import java.nio.file.Path;
-
+import com.amazon.opendistroforelasticsearch.security.auditlog.AuditLog;
```
Comment (vrozov) [inline src/main/java/com/amazon/opendistroforelasticsearch/security/dlic/rest/api/RolesMappingApiAction.java:18]: nit: avoid import reorder.

File: src/main/java/com/amazon/opendistroforelasticsearch/security/dlic/rest/api/TenantsApiAction.java [hunk 10]
```diff
@@ -1,6 +1,8 @@
 package com.amazon.opendistroforelasticsearch.security.dlic.rest.api;
 
 import java.nio.file.Path;
+import java.util.Arrays;
```
Comment (vrozov) [inline src/main/java/com/amazon/opendistroforelasticsearch/security/dlic/rest/api/TenantsApiAction.java:4]: nit: unused import.

File: src/main/java/com/amazon/opendistroforelasticsearch/security/dlic/rest/api/OpenDistroSecurityConfigAction.java [hunk 11]
```diff
@@ -58,21 +70,14 @@ public OpenDistroSecurityConfigAction(final Settings settings, final Path config
...
 
     @Override
-    protected void registerHandlers(RestController controller, Settings settings) {
-        controller.registerHandler(Method.GET, "/_opendistro/_security/api/securityconfig/", this);
-
-        //controller.registerHandler(Method.GET, "/_opendistro/_security/api/config/", this);
-
+    public List<Route> routes() {
         boolean enablePutOrPatch = settings.getAsBoolean(ConfigConstants.OPENDISTRO_SECURITY_UNSUPPORTED_RESTAPI_ALLOW_SECURITYCONFIG_MODIFICATION, false);
         if (enablePutOrPatch) {
```
Comment (vrozov) [inline src/main/java/com/amazon/opendistroforelasticsearch/security/dlic/rest/api/OpenDistroSecurityConfigAction.java:75]: use `allowPutOrPatch`

File: src/main/java/com/amazon/opendistroforelasticsearch/security/dlic/rest/api/OpenDistroSecurityConfigAction.java [hunk 12]
```diff
@@ -58,21 +70,14 @@ public OpenDistroSecurityConfigAction(final Settings settings, final Path config
...
-
-            //deprecated, will be removed with ODFE 8, use opendistro_security_config instead of config
-            controller.registerHandler(Method.PUT, "/_opendistro/_security/api/securityconfig/{name}", this);
-            controller.registerHandler(Method.PATCH, "/_opendistro/_security/api/securityconfig/", this);
+            return routes;
         }
+        return routes.stream().filter(route -> route.getMethod() == Method.GET).collect(Collectors.toList());
```
Comment (vrozov) [inline src/main/java/com/amazon/opendistroforelasticsearch/security/dlic/rest/api/OpenDistroSecurityConfigAction.java:78]: Create new `List` using `Collections.singletonList()`.

File: src/main/java/com/amazon/opendistroforelasticsearch/security/dlic/rest/api/OpenDistroSecurityConfigAction.java (hunk 11 above)
Comment (sujithvm) [reply src/main/java/com/amazon/opendistroforelasticsearch/security/dlic/rest/api/OpenDistroSecurityConfigAction.java:75]: Hmm. Wondering how that got there

**Comment by vrozov** [comment]
Is it necessary to upgrade version to v1.8.0.0?

**Comment by sujithvm** [comment]
Waiting on confirmation regarding the next version. 

I was thinking the version bump could be done in a separate PR/commit which can be tagged along with release

**Comment by vrozov** [comment]
ES version upgrade is an incompatible change and should trigger version change, IMO.

**Pull request merged**